sample-code/vc-using-framestream/vc-using-framestream/Connection.h

```diff
@@ -0,0 +1,58 @@
+#pragma once
+
+#include "commands.h"
+#include "AcousticSettings.h"
+#include "FrameStreamListener.h"
+#include "UdpListener.h"
+#include <memory>
+#include <boost/asio.hpp>
+
+namespace aris { class Command; }
+
+class Connection
+{
+public:
+  // Creates the Connection object with a live connection to the ARIS.
+  // Returns an unassociated unique_ptr if the connection could not be made.
+  static std::unique_ptr<Connection> Create(
+    boost::asio::io_service & io, 
+    const boost::asio::ip::tcp::endpoint & ep,
+    std::function<void(Aris::Network::FrameBuilder&)> onFrameCompletion,
+    SystemType systemType,
+    aris::Command::SetSalinity::Salinity salinity,
+    float initialFocusRange,
+    std::string & errorMessage);
+
+  // Use Create() rather than the ctor. Caller gives up ownership of commandSocket.
+  Connection(
+    boost::asio::io_service & io,
+    boost::asio::ip::tcp::socket && commandSocket,
+    boost::asio::ip::address receiveFrom,
+    std::function<void(Aris::Network::FrameBuilder&)> onFrameCompletion,
+    SystemType systemType,
+    aris::Command::SetSalinity::Salinity salinity,
+    float initialFocusRange);
+  ~Connection();
+
+  void SendCommand(const aris::Command & cmd);
```

Comment:
`Connection::Create()` handles such conditions and returns an unassociated unique_ptr along with an error message.